e2e/README.md

```diff
@@ -0,0 +1,366 @@
+
+## Prerequisites
+
+Before we get started there a few requirements.
+
+### Kubernetes Cluster Environment
+
+Your cluster must:
+
+- Be at least version 1.9
+- Have access to an S3-compatible object store ([Amazon S3](https://aws.amazon.com/s3/), [Google Storage](https://cloud.google.com/storage/docs/interoperability), [Minio](https://www.minio.io/kubernetes.html))
+- Contain 3 nodes of at least 8 cores and 16 GB of RAM.
+
+If using GKE, the following will provision a cluster with the required features:
+
+```
+export CLOUDSDK_CONTAINER_USE_CLIENT_CERTIFICATE=True
+gcloud alpha container clusters create ${USER} --enable-kubernetes-alpha --machine-type=n1-standard-8 --num-nodes=3 --disk-size=200 --zone=us-west1-a --cluster-version=1.9.2-gke.1 --image-type=UBUNTU
+```
+
+NOTE: You must be a Kubernetes admin to follow this guide. If you are not an admin, please contact your local cluster administrator for a client cert, or credentials to pass into the following commands:
+
+```
+$ kubectl config set-credentials <username> --username=<admin_username> --password=<admin_password>
+$ kubectl config set-context <context_name> --cluster=<cluster_name> --user=<username> --namespace=<namespace>
+$ kubectl config use-context <context_name>
+```
+
+### Local Setup
+
+You also need the following command line tools:
+
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- [argo](https://github.com/argoproj/argo/blob/master/demo.md#1-download-argo)
+- [helm](https://docs.helm.sh/using_helm/#installing-helm)
+- [ksonnet](https://ksonnet.io/#get-started)
+- [aws](https://docs.aws.amazon.com/cli/latest/userguide/installing.html)
+- [minio client](https://github.com/minio/mc#macos)
+
+NOTE: These instructions rely on Github, and may cause issues if behind a firewall with many Github users. Make sure you [generate and export this token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/):
+
+```
+export GITHUB_TOKEN=xxxxxxxx
+```
+
+## Modifying existing examples
+
+Many examples online use containers with pre-canned data, or scripts with certain assumptions as to the cluster spec. We will modify one of these [examples](https://github.com/tensorflow/tensorflow/tree/0375ffcf83e16c3d6818fa67c9c13de810c1dacf/tensorflow/tools/dist_test) to work with the tensorflow operator, and to work more like a real-world example.
+
+### Prepare model
+
+There is a delta between existing distributed mnist examples and what's needed to run well as a TFJob. These changes can be viewed in the [included diff](mnist-changes.md)
+
+Basically, we must
+
```

Comment:
Where is the original code? Why are these changes needed? Does the original example not support distributed?

A master isn't required; you can use worker 0 as the master which is what a lot of TF programs do.

Reply:
Looks like I forgot to commit the original! I'll get rid of the master bit, as you're right, we didn't end up needing it. This is more of an artifact of when we were trying to get master to work.

I'll also elaborate on the list of changes a bit.